bin/promalertproxy

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env perl
+
+use 5.028;
+use warnings;
+
+use Getopt::Long::Descriptive;
+use IO::Async::Loop;
+use PromAlertProxy::Hub;
+use PromAlertProxy::Config;
+
+my ($opt, $usage) = describe_options(
+  'promalertproxy -c config.toml',
+  [ 'config|c=s', 'config file to load', { required => 1 } ],
+  [ 'help',       'print usage message and exit', { shortcircuit => 1 } ],
+);
+
+print $usage->text and exit if $opt->help;
+
+my $loop = IO::Async::Loop->new;
+
+my $hub = PromAlertProxy::Hub->new(loop => $loop);
+
+PromAlertProxy::Config->new(hub => $hub, filename => $opt->config)->inflate;
```

Comment:
Yeah, I'm torn. I will explain and in the process convince myself one way or the other.

I like having the hub as a separate thing you add stuff to, because then you can add stuff to it from Perl. That seems to me like it makes tests "easier", and allows the thing to be embedded in something else.

On the other hand, its tough to incrementally upgrade an active object (`listen` speaks to that a little, I think: what if you call it twice?). Instantiating with a config object makes it a bit more like the old builder pattern; here's a description of what I want; make one.

(On the other other hand, being able to incrementally update the hub means you can do live reconfiguration, but that's not useful for this particular program. It is for Hello, which is where I first wrote this structure. That's just a case for making the right choice for the thing you're doing).

And of course, that config object doesn't need to be constructed out of a config _file_; you can make one of those straight from Perl; `::Config` could be a role in its own right, etc.

So I think, right now, I'm leaning slightly towards your way, but not so strongly that I'm in the mood to rewrite it right now. But Hello works this way, so I guess i should rethink that a bit. Though I need to rethink that a bit anyway if its every going to make it to production. Do you see why I want a real programmer?